src/Google.Pubsub.V1/PublisherClient.cs

```diff
@@ -318,36 +287,17 @@ public sealed partial class PublisherClientImpl : PublisherClient
                 "" // An empty page-token
             );
 
-        private readonly Func<DateTime?> _calcDeadline;
+        private readonly ClientHelper _clientHelper;
 
         public PublisherClientImpl(Publisher.IPublisherClient grpcClient, PublisherSettings settings)
         {
-            settings = settings ?? PublisherSettings.GetDefault();
             this.GrpcClient = grpcClient;
-            if (settings.Timeout == null)
-            {
-                this._calcDeadline = () => null;
-            }
-            else
-            {
-                IClock clock = settings.Clock ?? SystemClock.Instance;
-                TimeSpan timeout = settings.Timeout.Value;
-                this._calcDeadline = () => clock.GetCurrentDateTimeUtc() + timeout;
-            }
+            settings = settings ?? PublisherSettings.GetDefault();
+            _clientHelper = new ClientHelper(settings);
```

Comment:
Could make this a one-liner:
_clientHelper = new ClientHelper(settings ?? PublisherSettings.GetDefault());
(and in SubscriberClient)